src/troute-routing/troute/routing/diffusive_utils.py

```diff
@@ -502,6 +502,88 @@ def fp_naturalxsec_map(
     
     return x_bathy_g, z_bathy_g, mann_bathy_g, size_bathy_g, mxnbathy_g
 
+def fp_da_map(
+    mx_jorder,
+    ordered_reaches,
+    usgs_df,
+    nrch_g,
+    t0,
+    nsteps,
+    dt_da_g,
+    t0_g,
+    tfin_g):
+    """
+    Data assimilatoin data mapping between Python and Fortran
+    
+    Parameters
+    ----------
+    mx_jorder       -- (int) maximum network reach order
+    ordered_reaches -- (dict) reaches and reach metadata by junction order
+    usgs_df         -- (DataFrame) usgs streamflow data interpolated at dt time steps
+    t0              -- (datetime) initial date
+    nsteps          -- (int)  
+    dt_da_g         -- (int) numer of Data Assimilation timesteps
+    t0_g            -- (float) diffusive model's initial simulation time [hr] (by default, zero)
+    tfin_g          -- (float) diffusive model's final simulation time [hr] 
+    
+    Returns
+    -------
+    nts_da_g        -- (int) number of DA timesteps
+    usgs_da_g       -- (float) usgs oberved streamflow data [cms]
+    usgs_da_reach_g -- (int) indices of stream reaches where DA is applied    
+    """
+    from datetime import datetime, timedelta
+    import pandas as pd
```

Comment:
not good practice to import libraries in functions. This should be at the top of the script. Hard to believe that we don't already have pandas imported, here?